React/Base/RCTTouchHandler.m

```diff
@@ -124,7 +124,9 @@ - (void)_recordNewTouches:(NSSet *)touches
       continue;
     }
 #else // [TODO(macOS ISS#2323203)
-    CGPoint touchLocation = [self.view convertPoint:touch.locationInWindow fromView:nil];
+    // -[NSView hitTest:] takes coordinates in a view's superview coordinate system.
+    // The assumption here is that a RCTUIView/RCTSurfaceView will always have a superview.
+    CGPoint touchLocation = [self.view.superview convertPoint:touch.locationInWindow fromView:nil];
```

Comment:
I suspect this issue never arose before because previously the only superview (RCTRootView) would have the same coordinate system.